google/generativeai/retriever.py

```diff
@@ -0,0 +1,229 @@
+    client: glm.RetrieverServiceAsyncClient | None = None,
+) -> Corpus:
+    if client is None:
+        client = get_default_retriever_async_client()
+
+    if not name and not display_name:
+        raise ValueError("Either the corpus name or display name must be specified.")
+
+    corpus = None
+    if name:
+        if re.match(_CORPORA_NAME_REGEX, name):
+            corpus = glm.Corpus(name=name, display_name=display_name)
+        elif "corpora/" not in name:
+            corpus_name = "corpora/" + re.sub(_PATTERN, "", name)
+            corpus = glm.Corpus(name=corpus_name, display_name=display_name)
+        else:
+            raise ValueError("Corpus name must be formatted as corpora/<corpus_name>.")
+
+    request = glm.CreateCorpusRequest(corpus=corpus)
+    response = await client.create_corpus(request)
+    response = type(response).to_dict(response)
+    idecode_time(response, "create_time")
+    idecode_time(response, "update_time")
+    response = Corpus(**response)
+    return response
+
+
+def get_corpus(name: str, client: glm.RetrieverServiceClient | None = None) -> Corpus:
+    """
+    Get information about a specific `Corpus`.
+
+    Args:
+        name: The `Corpus` name.
+
+    Return:
+        `Corpus` of interest.
+    """
+    if client is None:
+        client = get_default_retriever_client()
+
+    request = glm.GetCorpusRequest(name=name)
+    response = client.get_corpus(request)
+    response = type(response).to_dict(response)
+    idecode_time(response, "create_time")
+    idecode_time(response, "update_time")
+    response = Corpus(**response)
+    return response
+
+
+@string_utils.set_doc(get_corpus.__doc__)
+async def get_corpus_async(
+    name: str, client: glm.RetrieverServiceAsyncClient | None = None
+) -> Corpus:
+    if client is None:
+        client = get_default_retriever_async_client()
+
+    request = glm.GetCorpusRequest(name=name)
+    response = await client.get_corpus(request)
+    response = type(response).to_dict(response)
+    idecode_time(response, "create_time")
+    idecode_time(response, "update_time")
+    response = Corpus(**response)
+    return response
+
+
+def delete_corpus(name: str, force: bool, client: glm.RetrieverServiceClient | None = None):
+    """
+    Delete a `Corpus`.
+
+    Args:
+        name: The `Corpus` name.
+        force: If set to true, any `Document`s and objects related to this `Corpus` will also be deleted.
+    """
+    if client is None:
+        client = get_default_retriever_client()
+
+    request = glm.DeleteCorpusRequest(name=name, force=force)
+    response = client.delete_corpus(request)
+
+
+@string_utils.set_doc(delete_corpus.__doc__)
+async def delete_corpus_async(
+    name: str, force: bool, client: glm.RetrieverServiceAsyncClient | None = None
+):
+    if client is None:
+        client = get_default_retriever_async_client()
+
+    request = glm.DeleteCorpusRequest(name=name, force=force)
+    response = await client.delete_corpus(request)
```

Comment:
WHy catch response and not return it?

Reply:
Removed `response` on all delete mehtods.